packages/p2p-v2/src/components/Modals/ShareAdsModal/ShareAdsModal.tsx

```diff
@@ -0,0 +1,158 @@
+import { customStyles } from '../helpers';
+import ShareMyAdsCard from './ShareAdsCard';
+import ShareMyAdsSocials from './ShareAdsSocials';
+import './ShareAdsModal.scss';
+
+type TShareAdsModalProps = {
+    id: string;
+    isModalOpen: boolean;
+    onRequestClose: () => void;
+};
+
+const websiteUrl = () => `${location.protocol}//${location.hostname}/`;
+
+const ShareAdsModal = ({ id, isModalOpen, onRequestClose }: TShareAdsModalProps) => {
+    const { isMobile } = useDevice();
+    const { data: advertInfo, isLoading: isLoadingInfo } = p2p.advert.useGet({ id });
+    const [isCopied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+    const { account_currency, advertiser_details, local_currency, rate_display, rate_type, type } = advertInfo ?? {};
+    const { id: advertiserId } = advertiser_details ?? {};
+
+    const divRef = useRef<HTMLDivElement | null>(null);
+    const advertUrl = `${websiteUrl()}cashier/p2p/advertiser?id=${advertiserId}&advert_id=${id}`;
+    const isBuyAd = type === BUY_SELL.BUY;
+    const firstCurrency = isBuyAd ? local_currency : account_currency;
+    const secondCurrency = isBuyAd ? account_currency : local_currency;
+    const adRateType = rate_type === RATE_TYPE.FLOAT ? '%' : ` ${local_currency}`;
+    const customMessage = `Hi! I'd like to exchange ${firstCurrency} for ${secondCurrency} at ${rate_display}${adRateType} on Deriv P2P.nnIf you're interested, check out my ad 👉nn${advertUrl} nnThanks!`;
+
+    const formattedMessage = customMessage.replace(/nn/g, '\n\n');
+
+    useEffect(() => {
+        Modal.setAppElement('#v2_modal_root');
+    }, []);
+
+    const onCopy = (event: { stopPropagation: () => void }) => {
+        copyToClipboard(advertUrl);
+        setIsCopied(true);
+        event.stopPropagation();
+    };
+
+    const handleGenerateImage = async () => {
+        if (divRef.current) {
+            const p2pLogo = divRef.current.querySelector('.p2p-v2-share-ads-card__qr-icon');
+            if (p2pLogo) {
+                const canvas = await html2canvas(divRef.current, { allowTaint: true, useCORS: true });
+                const screenshot = canvas.toDataURL('image/png', 1.0);
+                const fileName = `${type}_${id}.png`;
+                const link = document.createElement('a');
+                link.download = fileName;
+                link.href = screenshot;
+                link.click();
+            }
+        }
+    };
+
+    const handleShareLink = () => {
+        navigator.share({
+            text: formattedMessage,
+        });
+    };
+
+    useEffect(() => {
+        let timeoutClipboard: ReturnType<typeof setTimeout>;
+        if (isCopied) {
+            timeoutClipboard = setTimeout(() => {
+                setIsCopied(false);
+            }, 2000);
+        }
+        return () => {
+            clearTimeout(timeoutClipboard);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@nada-deriv, why could you be suppressing this? 🤔  
```suggestion
        // eslint-disable-next-line react-hooks/exhaustive-deps
```